src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexChannelFactoryTest.cs

```diff
@@ -0,0 +1,384 @@
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new NetTcpBinding();
+        Assert.Throws<ArgumentException>(() =>
+        {
+            DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, "http://not-the-right-scheme");
+            factory.CreateChannel();
+        });
+    }
+
+    [Fact]
+    // valid address, but the scheme is incorrect
+    public static void CreateChannel_ExpectedNetTcpScheme_InvalidScheme_ThrowsUriFormat()
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new NetTcpBinding();
+        Assert.Throws<ArgumentException>(() =>
+        {
+            DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, "qwerty://not-the-right-scheme");
+            factory.CreateChannel();
+        });
+    }
+
+    [Fact]
+    // valid address, but the scheme is incorrect
+    public static void CreateChannel_BasicHttpBinding_Throws_InvalidOperation()
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+
+        // Contract requires Duplex, but Binding 'BasicHttpBinding' doesn't support it or isn't configured properly to support it.
+        var exception = Assert.Throws<InvalidOperationException>(() =>
+        {
+            DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, "http://basichttp-not-duplex");
+            factory.CreateChannel();
+        });
+
+        Assert.Contains("BasicHttpBinding", exception.Message, StringComparison.OrdinalIgnoreCase);
+    }
+
+    [Fact]
+    public static void CreateChannel_Address_NullString_ThrowsArgumentNull()
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new NetTcpBinding();
+        Assert.Throws<ArgumentNullException>(() =>
+        {
+            DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, (string)null);
+        });
+    }
+
+    [Fact]
+    public static void CreateChannel_Address_NullEndpointAddress_ThrowsArgumentNull()
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new NetTcpBinding();
+        DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, (EndpointAddress)null);
+
+        Assert.Throws<InvalidOperationException>(() =>
+        {
+            factory.CreateChannel(); 
+        });
+    }
+
+    [Fact(Skip = "SecureConnectionDuplexSession is not supported yet.")]
+    public static void CreateChannel_Using_NetTcpBindingDefaults()
```

Comment:
This should be [ActiveIssue(nnn)] with the open issue for this.  Otherwise this test will fail in our CoreCLR/NETNative test environment.